src/main/kotlin/org/elm/ide/inspections/MissingCaseBranchAdder.kt

```diff
@@ -0,0 +1,86 @@
+package org.elm.ide.inspections
+
+import org.elm.lang.core.psi.ElmFile
+import org.elm.lang.core.psi.ElmPsiFactory
+import org.elm.lang.core.psi.ancestors
+import org.elm.lang.core.psi.elements.ElmAnythingPattern
+import org.elm.lang.core.psi.elements.ElmCaseOfExpr
+import org.elm.lang.core.psi.elements.ElmUnionPattern
+import org.elm.lang.core.types.TyUnion
+import org.elm.lang.core.types.findInference
+import org.elm.lang.core.types.renderParam
+
+/**
+ * This class can detect missing branches for case expressions and insert them into the PSI in place.
+ */
+class MissingCaseBranchAdder(val element: ElmCaseOfExpr) {
+    // This should be a lazy {}, but using it causes a compilation error due to conflicting
+    // declarations.
+    val missingBranches: List<TyUnion.Member>
+        get() {
+            if (_missingBranches == null) _missingBranches = calcMissingBranches()
+            return _missingBranches!!
+        }
+    private var _missingBranches: List<TyUnion.Member>? = null
+
+    fun addMissingBranches() {
+        if (missingBranches.isEmpty()) return
+
+        val patterns = missingBranches.map { b ->
+            (listOf(b.name) + b.parameters.map { it.renderParam() }).joinToString(" ")
+        }
+
+        addPatterns(patterns)
+    }
+
+    fun addWildcardBranch() {
+        if (missingBranches.isEmpty()) return
+        addPatterns(listOf("_"))
+    }
+
+    private fun addPatterns(patterns: List<String>) {
+        val factory = ElmPsiFactory(element.project)
+        val existingBranches = element.branches
+        val indent = "    ".repeat(element.ancestors.takeWhile { it !is ElmFile }.count())
```

Comment:
Clever. There are probably cases where this will break, but this is a great solution in lieu of doing the full code-formatter integration.